include/array/array.h

```diff
@@ -2174,26 +2172,6 @@ class array_ref {
   NDARRAY_HOST_DEVICE index_t rows() const { return shape_.rows(); }
   NDARRAY_HOST_DEVICE index_t columns() const { return shape_.columns(); }
 
-  /** Compare the contents of this array_ref to `other`. For two array_refs to
-   * be considered equal, they must have the same shape, and all elements
-   * addressable by the shape must also be equal. */
-  // TODO: Maybe this should just check for equality of the shape and pointer,
-  // and let the free function equal serve this purpose.
-  NDARRAY_HOST_DEVICE bool operator!=(const array_ref& other) const {
-    if (shape_ != other.shape_) { return true; }
-
-    // TODO: This currently calls operator!= on all elements of the array_ref,
-    // even after we find a non-equal element
-    // (https://github.com/dsharlet/array/issues/4).
-    bool result = false;
-    copy_shape_traits<Shape, Shape>::for_each_value(
-        shape_, base_, other.shape_, other.base_, [&](const_reference a, const_reference b) {
-          if (a != b) { result = true; }
-        });
-    return result;
-  }
-  NDARRAY_HOST_DEVICE bool operator==(const array_ref& other) const { return !operator!=(other); }
```

Comment:
Does this build? `array::operator==` uses this.

It looks like travis stopped working (again).